### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Sondro**

### CODE REVIEW

File: src/arrays.js [hunk 1]
```diff
@@ -4,49 +4,152 @@
...
 
 const reduce = (elements, cb, startingValue) => {
-  // Combine all elements into a single value going from left to right.
-  // Elements will be passed one by one into `cb`.
-  // `startingValue` is the starting value.  If `startingValue` is undefined then make `elements[0]` the initial value.
+ // Combine all elements into a single value going from left to right.
+ // Elements will be passed one by one into `cb`.
+ // `startingValue` is the starting value.  If `startingValue` is undefined then make `elements[0]` the initial value.
+ startingValue = elements[0];
+
```
Comment (sperrye) [inline src/arrays.js:47]: Be careful you're not overwriting an existing startingValue. You only want to set it to the first element in the array if it's undefined.

File: src/callbacks.js [hunk 2]
```diff
@@ -19,65 +19,124 @@
...
 
 // Write a function called contains that checks if an item is present inside of the given array.
 // Pass true to the callback if it is, otherwise pass false
 // code here
 
+const contains = (foods, str, cb) => {
+
+ let result = false;
+
+ for (let i = 0; i < foods.length; i++) {
```
Comment (sperrye) [inline src/callbacks.js:89]: Double check line 90. `[each]`

File: src/callbacks.js [hunk 3]
```diff
@@ -19,65 +19,124 @@
...
   console.log(result ? 'ribeye is in the array' : 'ribeye is not in the array');
 });
+*/
 
 // Write a function called removeDuplicates that removes all duplicate values from the given array.
 // Pass the array to the callback function.  Do not mutate the original array.
 // code here
 
+const removeDuplicates = (foods, cb) => {
+ let i = 1;
```
Comment (sperrye) [inline src/callbacks.js:107]: You don't need to initialize i, you do so in your for loop setup.

File: src/callbacks.js [hunk 4]
```diff
@@ -19,65 +19,124 @@
...
+   for (let i = 0; i < foods.length; i++) {
+    if (foods[i] === foods[i]) {
+     foods[i] = '';
+     i = 1;
+    }
+   }
+   i++;
+  }
+ }
+ cb(foods);
```
Comment (sperrye) [inline src/callbacks.js:119]: Review this function. If you nest for loops, you'll want to use different variable names for the indexes. Right now they're both set to i. You can also try writing it this way: initialize an empty array that will hold unique items. Set up a single for loop to iterate over the passed in array (which will be foods). For each item in the array, determine if it's in the `unique` array. You can use `indexOf` for this. If unique.indexOf(elements[i]) === -1, that means it's not in the unique array, and then you can push that element to the unique array. Finally when the for loop is finished, return the unique array.

File: src/objects.js [hunk 5]
```diff
@@ -1,26 +1,77 @@
 // Complete the following underscore functions.
 // Reference http://underscorejs.org/ for examples.
-
 const keys = (obj) => {
   // Retrieve all the names of the object's properties.
   // Return the keys as strings in an array.
   // Based on http://underscorejs.org/#keys
+  // console.log('Names: ' + Object.getOwnPropertyNames(obj));
+  return Object.keys(obj).toString();
```
Comment (sperrye) [inline src/objects.js:8]: You want to return the keys in an array, you won't need to apply `.toString()` here.

File: src/objects.js [hunk 6]
```diff
@@ -1,26 +1,77 @@
...
+*/
 
 const values = (obj) => {
   // Return all of the values of the object's own properties.
   // Ignore functions
   // http://underscorejs.org/#values
+  const arr = [];
+  const len = Object.values(obj).length;
+  let i = 0;
+
```
Comment (sperrye) [inline src/objects.js:24]: `Object.values(obj)` returns an array of the values in `obj`. You don't need to do anything else, just return this.

**Review by sperrye — COMMENTED** [review]
Completed review. If you have extra time consider re-working objects after reading about Object methods here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object

File: src/objects.js (hunk 6 above)
Comment (Sondro) [reply src/objects.js:24]: It says 'Ignore functions'

**Pull request closed without merging**